mypy/checker.py

```diff
@@ -2560,6 +2624,8 @@ def is_valid_inferred_type(typ: Type) -> bool:
     """
     if is_same_type(typ, NoneTyp()):
         return False
```

Comment:
Might add a comment here about how with strict Optional checking we'll infer `NoneTyp` later (in `leave_partial_types` I think) if we can't infer a specific `Optional` type by then, but we don't want to commit to `NoneTyp` now.